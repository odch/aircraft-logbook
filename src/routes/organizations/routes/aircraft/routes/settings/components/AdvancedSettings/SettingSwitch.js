import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core'
import ListItemText from '@material-ui/core/ListItemText'
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction'
import CircularProgress from '@material-ui/core/CircularProgress'
import Switch from '@material-ui/core/Switch'
import ListItem from '@material-ui/core/ListItem'
import Tooltip from '@material-ui/core/Tooltip'

const styles = {
  loadingIndicator: {
    verticalAlign: 'middle',
    height: '0.85em',
    marginRight: '0.1em'
  }
}

const SettingSwitch = ({
  label,
  checked,
  submitting,
  disabled,
  tooltip,
  onChange,
  classes
}) => (
  <ListItem disableGutters>
    <ListItemText primary={label} />
    <ListItemSecondaryAction>
      {submitting && (
        <CircularProgress size={16} className={classes.loadingIndicator} />
      )}
      <Tooltip
        title={tooltip || ''}
        disableFocusListener={!tooltip}
        disableHoverListener={!tooltip}
        disableTouchListener={!tooltip}
      >
        <span>
          <Switch
            edge="end"
            onChange={e => onChange(e.target.checked)}
            checked={checked}
            disabled={disabled || submitting}
          />
        </span>
      </Tooltip>
    </ListItemSecondaryAction>
  </ListItem>
)

SettingSwitch.propTypes = {
  label: PropTypes.string.isRequired,
  checked: PropTypes.bool.isRequired,
  submitting: PropTypes.bool,
  disabled: PropTypes.bool,
  tooltip: PropTypes.string,
  classes: PropTypes.object.isRequired,
  onChange: PropTypes.func.isRequired
}

export default withStyles(styles)(SettingSwitch)
