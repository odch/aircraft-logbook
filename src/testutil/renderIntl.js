import React from 'react'
import renderer from 'react-test-renderer'
import { createMount } from '@material-ui/core/test-utils'
import { IntlProvider } from 'react-intl'
import pretty from 'pretty'
import messages from '../messages'

const LOCALE = 'de'

const renderInIntlProvider = component => (
  <IntlProvider locale={LOCALE} messages={messages[LOCALE]}>
    {component}
  </IntlProvider>
)

const renderIntl = (component, options) =>
  renderer.create(renderInIntlProvider(component), options)

export const renderIntlUpdate = (renderer, component, options) =>
  renderer.update(renderInIntlProvider(component), options)

export const renderIntlMaterial = (component, returnPrettyHtml = false) => {
  const rendered = createMount()(renderInIntlProvider(component))
  return returnPrettyHtml ? pretty(rendered.html()) : rendered
}

export default renderIntl
