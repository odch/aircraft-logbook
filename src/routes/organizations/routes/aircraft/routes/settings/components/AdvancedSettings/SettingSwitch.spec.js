import React from 'react'
import renderIntl from '../../../../../../../../testutil/renderIntl'
import SettingSwitch from './SettingSwitch'

describe('routes', () => {
  describe('organizations', () => {
    describe('routes', () => {
      describe('aircraft', () => {
        describe('routes', () => {
          describe('settings', () => {
            describe('components', () => {
              describe('AdvancedSettings', () => {
                describe('SettingSwitch', () => {
                  it('renders correctly unchecked', () => {
                    const tree = renderIntl(
                      <SettingSwitch
                        label="My test switch"
                        checked={false}
                        onChange={() => {}}
                      />
                    ).toJSON()
                    expect(tree).toMatchSnapshot()
                  })

                  it('renders correctly checked', () => {
                    const tree = renderIntl(
                      <SettingSwitch
                        label="My test switch"
                        checked={true}
                        onChange={() => {}}
                      />
                    ).toJSON()
                    expect(tree).toMatchSnapshot()
                  })

                  it('renders switch readonly with loading indicator when submitting', () => {
                    const tree = renderIntl(
                      <SettingSwitch
                        label="My test switch"
                        checked={true}
                        submitting
                        onChange={() => {}}
                      />
                    ).toJSON()
                    expect(tree).toMatchSnapshot()
                  })
                })
              })
            })
          })
        })
      })
    })
  })
})
