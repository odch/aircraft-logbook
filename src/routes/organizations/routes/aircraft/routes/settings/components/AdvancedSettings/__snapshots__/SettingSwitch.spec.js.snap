// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes organizations routes aircraft routes settings components AdvancedSettings SettingSwitch renders correctly checked 1`] = `
<li
  className="MuiListItem-container"
>
  <div
    className="MuiListItem-root MuiListItem-secondaryAction"
    disabled={false}
  >
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
      >
        My test switch
      </span>
    </div>
  </div>
  <div
    className="MuiListItemSecondaryAction-root"
  >
    <span
      className="MuiSwitch-root MuiSwitch-edgeEnd"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-62 MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-checked-63 Mui-checked"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <span
            className="MuiSwitch-thumb"
          />
          <input
            checked={true}
            className="PrivateSwitchBase-input-65 MuiSwitch-input"
            onChange={[Function]}
            type="checkbox"
          />
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </span>
      <span
        className="MuiSwitch-track"
      />
    </span>
  </div>
</li>
`;

exports[`routes organizations routes aircraft routes settings components AdvancedSettings SettingSwitch renders correctly unchecked 1`] = `
<li
  className="MuiListItem-container"
>
  <div
    className="MuiListItem-root MuiListItem-secondaryAction"
    disabled={false}
  >
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
      >
        My test switch
      </span>
    </div>
  </div>
  <div
    className="MuiListItemSecondaryAction-root"
  >
    <span
      className="MuiSwitch-root MuiSwitch-edgeEnd"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-62 MuiSwitch-switchBase MuiSwitch-colorSecondary"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <span
            className="MuiSwitch-thumb"
          />
          <input
            checked={false}
            className="PrivateSwitchBase-input-65 MuiSwitch-input"
            onChange={[Function]}
            type="checkbox"
          />
        </span>
      </span>
      <span
        className="MuiSwitch-track"
      />
    </span>
  </div>
</li>
`;

exports[`routes organizations routes aircraft routes settings components AdvancedSettings SettingSwitch renders switch readonly with loading indicator when submitting 1`] = `
<li
  className="MuiListItem-container"
>
  <div
    className="MuiListItem-root MuiListItem-secondaryAction"
    disabled={false}
  >
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
      >
        My test switch
      </span>
    </div>
  </div>
  <div
    className="MuiListItemSecondaryAction-root"
  >
    <div
      className="MuiCircularProgress-root SettingSwitch-loadingIndicator-1 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style={
        Object {
          "height": 16,
          "width": 16,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
    <span
      className="MuiSwitch-root MuiSwitch-edgeEnd"
    >
      <span
        aria-disabled={true}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-62 MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-checked-63 Mui-checked PrivateSwitchBase-disabled-64 Mui-disabled Mui-disabled Mui-disabled"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
      >
        <span
          className="MuiIconButton-label"
        >
          <span
            className="MuiSwitch-thumb"
          />
          <input
            checked={true}
            className="PrivateSwitchBase-input-65 MuiSwitch-input"
            disabled={true}
            onChange={[Function]}
            type="checkbox"
          />
        </span>
      </span>
      <span
        className="MuiSwitch-track"
      />
    </span>
  </div>
</li>
`;
